src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1087,6 +1109,63 @@ emit_vector_insert_element (
 	return ins;
 }
 
+#if defined(TARGET_ARM64)
+static MonoInst*
+emit_normalize_vector_2_3_4 (MonoCompile *cfg, MonoClass *klass, MonoInst *arg){
```

Comment:
Calculating the reciprocal sqrt and then multiplying should lead to more efficient (literature states around 4x) solution than using division by `sqrt`.

Reply:
> Calculating the reciprocal sqrt and then multiplying should lead to more efficient (literature states around 4x) solution than using division by sqrt.

That is true. I was a bit afraid that it could change the behavior of the API, i.e., does it give the results users will expect.

Reply:
I see. I didn't see any differences between this and coreclr implementation. They both behave the same when provided 0 value and when provided values really close to 0. 

@matouskozak do you have any example in mind?

@fanyang-mono  what do you think, should we reimplement that using sqrt and division?

Reply:
`Sqrt` is strictly `Sqrt`, it is not safe to use an estimating function. The same goes for most APIs/scenarios.

We explicitly expose `*Estimate` functions (like `float.ReciprocalEstimate`) so users can opt into estimations and otherwise get correct/deterministic results across hardware/architectures.

Instructions like `FRSQRTE` can result in a massive amount of precision loss, especially across complex sequences of operations.